src/sdk/pynni/nni/smac_tuner/smac_tuner.py

```diff
@@ -0,0 +1,190 @@
+            self.__module__ + "." + self.__class__.__name__)
+        self.optimize_mode = OptimizeMode(optimize_mode)
+        self.total_data = {}
+        self.optimizer = None
+        self.smbo_solver = None
+        self.first_one = True
+        self.update_ss_done = False
+
+    def _main_cli(self):
+        '''
+        Main function of SMAC for CLI interface
+        '''
+        self.logger.info("SMAC call: %s" % (" ".join(sys.argv)))
+
+        cmd_reader = CMDReader()
+        args_, _ = cmd_reader.read_cmd()
+
+        root_logger = logging.getLogger()
+        root_logger.setLevel(args_.verbose_level)
+        logger_handler = logging.StreamHandler(
+                stream=sys.stdout)
+        if root_logger.level >= logging.INFO:
+            formatter = logging.Formatter(
+                "%(levelname)s:\t%(message)s")
+        else:
+            formatter = logging.Formatter(
+                "%(asctime)s:%(levelname)s:%(name)s:%(message)s",
+                "%Y-%m-%d %H:%M:%S")
+        logger_handler.setFormatter(formatter)
+        root_logger.addHandler(logger_handler)
+        # remove default handler
+        root_logger.removeHandler(root_logger.handlers[0])
+
+        # Create defaults
+        rh = None
+        initial_configs = None
+        stats = None
+        incumbent = None
+
+        # Create scenario-object
+        scen = Scenario(args_.scenario_file, [])
+
+        if args_.mode == "SMAC":
+            optimizer = SMAC(
+                scenario=scen,
+                rng=np.random.RandomState(args_.seed),
+                runhistory=rh,
+                initial_configurations=initial_configs,
+                stats=stats,
+                restore_incumbent=incumbent,
+                run_id=args_.seed)
+        elif args_.mode == "ROAR":
+            optimizer = ROAR(
+                scenario=scen,
+                rng=np.random.RandomState(args_.seed),
+                runhistory=rh,
+                initial_configurations=initial_configs,
+                run_id=args_.seed)
+        elif args_.mode == "EPILS":
+            optimizer = EPILS(
+                scenario=scen,
+                rng=np.random.RandomState(args_.seed),
+                runhistory=rh,
+                initial_configurations=initial_configs,
+                run_id=args_.seed)
+        else:
+            optimizer = None
+
+        return optimizer
+
+    def update_search_space(self, search_space):
+        '''
+        TODO: this is urgly, we put all the initialization work in this method, 
+        because initialization relies on search space, also because update_search_space is called at the beginning.
+        NOTE: updating search space is not supported.
+        '''
+        if not self.update_ss_done:
+            generate_scenario(search_space)
+            self.optimizer = self._main_cli()
+            self.smbo_solver = self.optimizer.solver
+            self.update_ss_done = True
+        else:
+            pass
```

Comment:
If it is not supported, should we raise exception instead of "pass"?

Reply:
add log here